src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/model/DestinationTests.kt

```diff
@@ -0,0 +1,93 @@
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.model
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.destination.Chime
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.destination.CustomWebhook
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.destination.Slack
+import org.elasticsearch.test.ESTestCase
+
+/**
+ * Temporary import from alerting, this will be removed once we pull notifications out of
+ * alerting so all plugins can consume and use.
+ */
+class DestinationTests : ESTestCase() {
+
+    fun `test chime destination`() {
+        val chime = Chime("http://abc.com")
+        assertEquals("Url is manipulated", chime.url, "http://abc.com")
+    }
+
+    fun `test chime destination with out url`() {
+        try {
+            Chime("")
+            fail("Creating a chime destination with empty url did not fail.")
+        } catch (ignored: IllegalArgumentException) {
+        }
+
+        try {
+            Chime(null)
+            fail("Creating a chime destination with null url did not fail.")
+        } catch (ignored: IllegalArgumentException) {
+        }
+    }
+
+    fun `test slack destination`() {
+        val slack = Slack("http://abc.com")
+        assertEquals("Url is manipulated", slack.url, "http://abc.com")
+    }
+
+    fun `test slack destination with out url`() {
+        try {
+            Slack("")
+            fail("Creating a slack destination with empty url did not fail.")
+        } catch (ignored: IllegalArgumentException) {
+        }
+
+        try {
+            Slack(null)
+            fail("Creating a slack destination with null url did not fail.")
+        } catch (ignored: IllegalArgumentException) {
+        }
+    }
+
+    fun `test custom webhook destination with url and no host`() {
+        val customWebhook = CustomWebhook("http://abc.com", null, null, -1, null, emptyMap(), emptyMap(), null, null)
+        assertEquals("Url is manipulated", customWebhook.url, "http://abc.com")
+    }
+
+    fun `test custom webhook destination with host and no url`() {
+        try {
+            val customWebhook = CustomWebhook(null, null, "abc.com", 80, null, emptyMap(), emptyMap(), null, null)
+            assertEquals("host is manipulated", customWebhook.host, "abc.com")
+        } catch (ignored: IllegalArgumentException) {
+        }
+    }
+
+    fun `test custom webhook destination with url and host`() {
```

Comment:
This test says ```with url and host``` but only a ```url``` was given

Reply:
This is just a copy from Alerting, it shouldn't live in this repo for long and will be removed so should be fine for now.